### PR TITLE
 virttest.utils_misc: add some steps about partition

### PR DESCRIPTION
**Pull request by maxujun**
virttest.utils_misc: add some steps about partition
id:1324316  1324309 1324320

### CODE REVIEW

**Comment by MiriamDeng** [comment]
Had better give a description about this PR .

**Comment by maxujun** [comment]
@MiriamDeng 
1.add steps of creating partition table
2.add step of getting partitions
3.format partition instead of whole disk 
4.mount partionn instead of whole disk

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -2756,14 +2756,25 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
...
+            list_parttable_cmd = "parted -s %s print" % kname
+            mklabel_cmd = "parted -s %s mklabel msdos" % kname
+            output = session.cmd_output(list_parttable_cmd)
+            if "Error" in output:
+                logging.info("Create partition table on disk '%s'" % kname)
+                session.cmd(mklabel_cmd)
             logging.info("Create partition on disk '%s'" % kname)
             mkpart_cmd = "parted -s %s  mkpart primary 0 %s" % (kname, size)
             session.cmd(mkpart_cmd)
+            time.sleep(5)
```
Comment (mazhang) [inline virttest/utils_misc.py]: 1. I think seesion.cmd() will got return when "mkport_cmd" complete, so sleep not necessary.
2. Better to execute "partprobe"  after change partition label.

Comment (maxujun) [inline virttest/utils_misc.py]: no，I have tested it ,it won't get partition infomation if no sleep time.

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -2756,14 +2756,25 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
     for disk in disks.splitlines():
         if did in disk:
             kname = disk.split()[0]
+            list_parttable_cmd = "parted -s %s print" % kname
+            mklabel_cmd = "parted -s %s mklabel msdos" % kname
+            output = session.cmd_output(list_parttable_cmd)
+            if "Error" in output:
```
Comment (xutian) [inline virttest/utils_misc.py]: do you mean not in.  if error in output, I think that's mean create partition failed.

Comment (xutian) [inline virttest/utils_misc.py]: See it, please move mklabel_cmd in if branch.

File: virttest/utils_misc.py (hunk 1 above)
Comment (xutian) [inline virttest/utils_misc.py]: five seconds is too long, better to use wait_for()

File: virttest/utils_misc.py [hunk 3]
```diff
@@ -2756,14 +2756,25 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
...
+                logging.info("Create partition table on disk '%s'" % kname)
+                session.cmd(mklabel_cmd)
             logging.info("Create partition on disk '%s'" % kname)
             mkpart_cmd = "parted -s %s  mkpart primary 0 %s" % (kname, size)
             session.cmd(mkpart_cmd)
+            time.sleep(5)
+            logging.info("Try to get the partitions of '%s'" % kname)
+            get_part_cmd = "readlink -f /dev/disk/by-id/*|grep %s\\\w" % kname
```
Comment (xutian) [inline virttest/utils_misc.py]: I'm not sure, this command can work in 2.6 kernel.

Comment (xutian) [inline virttest/utils_misc.py]: "partprobe  -s /dev/%s" % kname   can summary partition in the disk

**Comment by maxujun** [comment]
@xutian Have already updated,thanks for comments

File: virttest/utils_misc.py [hunk 4]
```diff
@@ -2756,14 +2756,25 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
...
+            if "Error" in output:
+                logging.info("Create partition table on disk '%s'" % kname)
+                mklabel_cmd = "parted -s %s mklabel msdos" % kname
+                session.cmd(mklabel_cmd)
             logging.info("Create partition on disk '%s'" % kname)
             mkpart_cmd = "parted -s %s  mkpart primary 0 %s" % (kname, size)
             session.cmd(mkpart_cmd)
+            time.sleep(2)
+            logging.info("Try to get the partitions of '%s'" % kname)
+            get_part_cmd = "partprobe -s /dev/sda|awk '{$1=$2=$3="";print}'"
```
Comment (luckyh) [inline virttest/utils_misc.py]: `/dev/sda` should be the value of `kname`.

File: virttest/utils_misc.py [hunk 5]
```diff
@@ -2756,14 +2756,25 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
...
+            parts = session.cmd(get_part_cmd).splitlines()
+            logging.info("The partitions are %s%s" % (kname, parts))
```
Comment (luckyh) [inline virttest/utils_misc.py]: I don't think it could get that you expected.

File: virttest/utils_misc.py [hunk 6]
```diff
@@ -2756,14 +2756,25 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
...
             logging.info("Format partition to '%s'" % fstype)
-            format_cmd = "yes|mkfs -t %s %s" % (kname, fstype)
+            format_cmd = "yes|mkfs  %s%s -t %s" % (kname, parts, fstype)
             session.cmd(format_cmd)
             logging.info("Mount the disk to '%s'" % mountpoint)
-            mount_cmd = "mount -t %s %s %s" % (fstype, kname, mountpoint)
+            mount_cmd = "mount -t %s%s %s %s" % (fstype, kname, parts, mountpoint)
```
Comment (luckyh) [inline virttest/utils_misc.py]: Should be:
```python
"mount -t %s %s%s %s"
```

**Comment by maxujun** [comment]
@luckyh Have already updated it,thanks.

**Comment by maxujun** [comment]
@luckyh Could you help review this patch?

File: virttest/utils_misc.py [hunk 7]
```diff
@@ -2755,14 +2755,26 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
...
+            list_parttable_cmd = "parted -s %s print" % kname
+            output = session.cmd_output(list_parttable_cmd)
+            if "Error" in output:
+                logging.info("Create partition table on disk '%s'" % kname)
+                mklabel_cmd = "parted -s %s mklabel msdos" % kname
+                session.cmd(mklabel_cmd)
             logging.info("Create partition on disk '%s'" % kname)
             mkpart_cmd = "parted -s %s  mkpart primary 0 %s" % (kname, size)
             session.cmd(mkpart_cmd)
+            time.sleep(2)
```
Comment (luckyh) [inline virttest/utils_misc.py]: This line can be removed because you have informed the OS of partition table changes via `partprobe` below.

**Comment by xutian** [comment]
Please provider ID list.

File: virttest/utils_misc.py [hunk 8]
```diff
@@ -2755,14 +2755,26 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
     for disk in disks.splitlines():
         if did in disk:
             kname = disk.split()[0]
+            list_parttable_cmd = "parted -s %s print" % kname
```
Comment (xutian) [inline virttest/utils_misc.py]: I think command should be like

`parted /dev/sda print`

File: virttest/utils_misc.py [hunk 9]
```diff
@@ -2755,14 +2755,26 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
...
+            output = session.cmd_output(list_parttable_cmd)
```
Comment (xutian) [inline virttest/utils_misc.py]: better to use check status instead output

File: virttest/utils_misc.py (hunk 8 above)
Comment (xutian) [inline virttest/utils_misc.py]: or parted /dev/sda -s print

File: virttest/utils_misc.py [hunk 10]
```diff
@@ -2755,14 +2755,26 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
...
+            if "Error" in output:
+                logging.info("Create partition table on disk '%s'" % kname)
+                mklabel_cmd = "parted -s %s mklabel msdos" % kname
```
Comment (xutian) [inline virttest/utils_misc.py]: parted  /tmp/zz.qcow2 -s "mklabel msdos"

Comment (xutian) [inline virttest/utils_misc.py]: please don't forget do a simple unit test after update.

Comment (maxujun) [inline virttest/utils_misc.py]: I don't understand "parted /tmp/zz.qcow2 -s "mklabel msdos"" ,do you mean can creat partition table to image file directly? I test it ，and I can't boot up my vm with error “Image is not in qcow2 format
”

File: virttest/utils_misc.py (hunk 8 above)
Comment (maxujun) [inline virttest/utils_misc.py]: Have already updated.thanks

File: virttest/utils_misc.py (hunk 9 above)
Comment (maxujun) [inline virttest/utils_misc.py]: Have already updated.thanks

File: virttest/utils_misc.py (hunk 10 above)
Comment (maxujun) [inline virttest/utils_misc.py]: Have already updated it as comment above.thanks

**Pull request closed without merging**